corpus/convert.py

```diff
@@ -93,6 +106,59 @@ def run(self):
         c = corpus.Corpus()
         c.load(corpus_path)
 
-        with open(self.txt_path.get_path(), "wt") as out:
-            for segment in c.segments():
+        if self.segment_file:
+            with open(tk.uncached_path(self.segment_file), "rt") as f:
+                lines = [l for l in f.readlines() if len(l.strip()) > 0]
```

Comment:
`lines = set(l for l in ...)` directly create the set from the iterator, no need for an intermediate list
also: why not strip the line that is added to the set?